### ISSUE TITLE
@Overload stubs are flagged with to many args(WPS211) and useless statement(WPS428)

### ISSUE THREAD

**Issue by maltevesper** [wemake-services/wemake-python-styleguide#1957]
### What's wrong

On Overload specificaitons of functions WPS211 and WPS428 is reported

```
from typing import overload


@overload
def func(a1, b1, c1, d1, e1, f1, g1):
    ...


def func(a1, b1, c1, d1, e1, f1, g1, h1):
    """Dummy."""
    ...
```

### How it should be

WPS211 and WPS428 should be silenced for function stubs marked with `@overload`.
The `useless` statement in the stub is needed (WPS428), and the too many args violation should only be flagged on the actual implementation (WPS211).

### Flake8 version and plugins

$ flake8 --bugreport
{
  "dependencies": [],
  "platform": {
    "python_implementation": "CPython",
    "python_version": "3.9.0",
    "system": "Windows"
  },
  "plugins": [
    {
      "is_local": false,
      "plugin": "flake8-bandit",
      "version": "2.1.2"
    },
    {
      "is_local": false,
      "plugin": "flake8-broken-line",
      "version": "0.3.0"
    },
    {
      "is_local": false,
      "plugin": "flake8-bugbear",
      "version": "21.3.2"
    },
    {
      "is_local": false,
      "plugin": "flake8-comprehensions",
      "version": "3.4.0"
    },
    {
      "is_local": false,
      "plugin": "flake8-darglint",
      "version": "1.7.0"
    },
    {
      "is_local": false,
      "plugin": "flake8-debugger",
      "version": "4.0.0"
    },
    {
      "is_local": false,
      "plugin": "flake8-docstrings",
      "version": "1.6.0, pydocstyle: 5.1.1"
    },
    {
      "is_local": false,
      "plugin": "flake8-eradicate",
      "version": "1.0.0"
    },
    {
      "is_local": false,
      "plugin": "flake8-pytest-style",
      "version": "1.4.0"
    },
    {
      "is_local": false,
      "plugin": "flake8-string-format",
      "version": "0.3.0"
    },
    {
      "is_local": false,
      "plugin": "flake8_commas",
      "version": "2.0.0"
    },
    {
      "is_local": false,
      "plugin": "flake8_isort",
      "version": "4.0.0"
    },
    {
      "is_local": false,
      "plugin": "flake8_quotes",
      "version": "3.2.0"
    },
    {
      "is_local": false,
      "plugin": "mccabe",
      "version": "0.6.1"
    },
    {
      "is_local": false,
      "plugin": "naming",
      "version": "0.11.1"
    },
    {
      "is_local": false,
      "plugin": "nitpick",
      "version": "0.26.0"
    },
    {
      "is_local": false,
      "plugin": "pycodestyle",
      "version": "2.7.0"
    },
    {
      "is_local": false,
      "plugin": "pyflakes",
      "version": "2.3.0"
    },
    {
      "is_local": false,
      "plugin": "rst-docstrings",
      "version": "0.0.14"
    },
    {
      "is_local": false,
      "plugin": "wemake_python_styleguide",
      "version": "0.15.2"
    }
  ],
  "version": "3.9.0"
}

### pip information

pip 21.0.1 from C:\Users\***\AppData\Local\pypoetry\Cache\virtualenvs\wemake-python-styleguide-VSx05pEp-py3.9\lib\site-packages\pip (python 3.9)
added-value @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/b2/68/48/d6ac1387cd772e3b40393648c152dfaadeec9323035c2b46bce584fe3f/added_value-0.15.0-py2.py3-none-any.whl
alabaster @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/48/57/b7/0e44080d4c6f80ece83e427f90368b376fdec42759b60c2c37a04da23b/alabaster-0.7.12-py2.py3-none-any.whl
asciitree @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/4f/3d/6e/a141cfba51c57886b5f40d1e8a107d8522417aa0092f7651e8541caffd/asciitree-0.3.3.tar.gz
astboom @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/84/60/5d/d613572a5390644e876fc3204f018865eace95eec46f2399cb39d4729c/astboom-0.4.2-py3-none-any.whl
astor @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/1f/03/36/982d1222edac5e8cb8ac6e0464249747fa800d4fb04728a99153ecfe4d/astor-0.8.1-py2.py3-none-any.whl
astpath @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/39/18/67/79f849dc5836320eb05696fda3fd3fcdf80f644dcc5939e31e26b6179c/astpath-0.9.0.tar.gz
atomicwrites @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/b7/99/9c/c8ddc18c8225c740fc2ae6d503da3f93d5dddaf04afac3da460afacbc6/atomicwrites-1.4.0-py2.py3-none-any.whl
attrs @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/f9/48/82/553e4bef24d3b294c0c18f27a7853f3ed151508efd144cb7ea37db1c48/attrs-20.3.0-py2.py3-none-any.whl
autopep8 @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/80/d5/2f/299852fb6ea9af809c0778983afa447fb2313b8cd80ead92319fe05ba6/autopep8-1.5.6-py2.py3-none-any.whl
autorepr @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/79/b0/2c/559adfad5f45b74ef658fdab7bff52f955339a62d1aeae798f390f88d9/autorepr-0.3.0-py2.py3-none-any.whl
Babel @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/d9/3d/cd/73e78690454f079f3f9a6cbb78d75b98f6b8b2f982e5a58854a91e6cfd/Babel-2.9.0-py2.py3-none-any.whl
backcall @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/43/8e/e8/4e598704edf6fb4a53d552ea511c04e9958dcf850897760e5387878b99/backcall-0.2.0-py2.py3-none-any.whl
bandit @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/6a/05/4f/98680ab175e4b595c2d1b775974c208b6b20c05448a52944374c2db4b0/bandit-1.7.0-py3-none-any.whl
cachy @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/c0/ce/2b/be65c61ed593659749cff39804b346369e358f14f09151abe591f0c5ab/cachy-0.3.0-py2.py3-none-any.whl
certifi @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/d8/df/24/ed696681f34f8916b0aef99138db9a94e37d54684b9829af34a7fd4e39/certifi-2020.12.5-py2.py3-none-any.whl
chardet @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/11/63/f9/797eda27963177a6b75a340f62aa194d462ea69e6b0dbb77a651fa2b62/chardet-4.0.0-py2.py3-none-any.whl
click @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/e2/79/34/a23e9d2f683ed66be11ec3bd760dec3a2fe228cfdedf2071bcf0531b06/click-7.1.2-py2.py3-none-any.whl
colorama @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/9e/b3/11/7d87ac44fdb2d557301f1f4086a37c080d1482a98751abe7cdbabbad26/colorama-0.4.4-py2.py3-none-any.whl
ConfigUpdater @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/49/be/2a/0c66ba41abfc4e2d944ab8aabaea9e43d2ec2e54ab849a5f424736e4f8/ConfigUpdater-2.0-py2.py3-none-any.whl
coverage @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/ca/08/1f/33c9eb49ec9c1a16fbd3ca0b112f0a90b91410de84d3f6bf62c55cc6a7/coverage-5.4-cp39-cp39-win_amd64.whl
coverage-conditional-plugin @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/0c/88/ac/39b935673d6e547332ca00ed580c16e0a1379d9b8a843ac84eaceb676e/coverage_conditional_plugin-0.3.1-py3-none-any.whl
darglint @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/cd/5b/5e/804102fc3c31c1f8d46079af65e0dc5460a746513ec4931be21febfd33/darglint-1.7.0-py3-none-any.whl
decorator @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/4e/0d/42/3677b6c493b68794c351e9fbd528ed0b596d4b11793d757285eb694b48/decorator-4.4.2-py2.py3-none-any.whl
dictdiffer @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/89/cd/10/56605467e905287a1404203b6fd708ad810132b18fa01a03f5e2db6d5e/dictdiffer-0.8.1-py2.py3-none-any.whl
doc8 @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/51/46/c8/750e726ab4a3544a2c01374dd6db52d420d86275cc183e16630f5e4fd3/doc8-0.8.1-py2.py3-none-any.whl
docutils @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/2e/41/32/0d23511bc6fb09a4386bbedfbb9a68a8e1fc5d502364be1b30e6af1f07/docutils-0.16-py2.py3-none-any.whl
dparse @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/0e/d3/3f/27b08502c8c8da888ca1b2701c4f1d91b6777be03197cc2566b65eb3dd/dparse-0.5.1-py3-none-any.whl
eradicate @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/12/ce/ac/197035fe6d51568abb7ea160f5ad416d2164a2010005e8356b8229e550/eradicate-2.0.0.tar.gz
fastdiff @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/a3/54/d1/2068014cf00b0b58dcbe0b8e6bd71a873b83c8b078532c947bc2555514/fastdiff-0.2.0.tar.gz
flake8 @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/09/d4/ba/f775b54656730d949029bb13567c7e05972fd4dce4ec8f780263878eae/flake8-3.9.0-py2.py3-none-any.whl
flake8-bandit @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/7e/e4/46/e15782d941f9cde39b64ca5b636180f47573f2b2c9315be56b55152f17/flake8_bandit-2.1.2.tar.gz
flake8-broken-line @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/51/ea/87/37348b281b73d7df44fc46b09c0430e2984e991df11998e2e9bb459fce/flake8_broken_line-0.3.0-py3-none-any.whl
flake8-bugbear @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/3b/53/77/a98850b6c5de076522833117754167d19d3231ce11c2a12412c8211a98/flake8_bugbear-21.3.2-py36.py37.py38-none-any.whl
flake8-commas @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/c3/47/1d/7f7fac0c58b2bd2bf7361bcba0bceba1c81c365cab5e1de352fa7fac68/flake8_commas-2.0.0-py2.py3-none-any.whl
flake8-comprehensions @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/e8/4d/4e/922afc66c264d167b1a218938d6ff1efd36562f903d9475b2228f3639e/flake8_comprehensions-3.4.0-py3-none-any.whl
flake8-debugger @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/66/04/47/7bef98a8d237eb17cbfbcb803343be1c79e2c0674ceba163717b6c8e1b/flake8_debugger-4.0.0-py3-none-any.whl
flake8-docstrings @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/e0/85/e9/6b482a11d48cf26e1170d9f5bf0b044a5a6c9b816ffe70945e90fc3e56/flake8_docstrings-1.6.0-py2.py3-none-any.whl
flake8-eradicate @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/99/8a/26/95fdb311c16f732e009ab3febcd098f7287b47ae77d547d51f8e69027e/flake8_eradicate-1.0.0-py3-none-any.whl
flake8-isort @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/11/37/1c/68fb64c6704b9c2468f711b83090590abc2c8295eeafbac9a167f32e0a/flake8_isort-4.0.0-py2.py3-none-any.whl
flake8-plugin-utils @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/05/01/93/d85d5e280ed56eaf6400f1215ca97460ff0d1aca89497cfc6a770d5b91/flake8_plugin_utils-1.3.1-py3-none-any.whl
flake8-polyfill @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/28/17/cc/952c11cd5ffb2608137557f928dc4f9365b4dbe1e2a6015eeea78583ac/flake8_polyfill-1.0.2-py2.py3-none-any.whl
flake8-pytest-style @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/1f/cc/cc/bc938a9c0a089cb6c1f46969bcb381291c4554c32da7f959f84734055b/flake8_pytest_style-1.4.0-py3-none-any.whl
flake8-quotes @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/75/61/73/b33ec4139bc79d01b0748fb1ae5889fbd6bd544c6f35521fb4dd981b1a/flake8-quotes-3.2.0.tar.gz
flake8-rst-docstrings @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/c1/17/4f/04f746759e0ed22c1aea9c886c90bf4f9ac24cc65c066cc7b6f40a6057/flake8-rst-docstrings-0.0.14.tar.gz
flake8-string-format @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/24/89/bb/7ce8e216f8c7289aa8a2ad4c44f30f87af6c7cdaf5d510110d566d66ec/flake8_string_format-0.3.0-py2.py3-none-any.whl
gitdb @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/d1/bc/72/319b4877a5ac3bdbf63b366837736220536a3c9b7fbec62597df7e16ab/gitdb-4.0.5-py3-none-any.whl
GitPython @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/dc/d0/3c/c0d1f2ae74170af3d1ef5f0fc41654e2d564e00c4efbf1e5e095e56f36/GitPython-3.1.13-py3-none-any.whl
grimp @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/51/22/77/df37e1c1ab6bbeae7c01cfc47a72af67fbb6f9d332f6f7c8dfca37004c/grimp-1.2.3.tar.gz
hypothesis @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/46/9a/3a/d2fc5b270124d9f2ddf1f04e1be1bf62af4544adadc937a6b3a1249e38/hypothesis-6.8.1-py3-none-any.whl
hypothesmith @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/29/a1/94/06e2a9a37346488207c5ad6e644a64481c82a8e481cb30c296bd64f7c3/hypothesmith-0.1.8-py3-none-any.whl
identify @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/65/41/3e/6f4289274bc34a5f5939e50de6607091a175e07e775da95da7c06a7c41/identify-1.5.13-py2.py3-none-any.whl
idna @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/ef/7f/a9/19cc0b8760bdf6f696290c06532496f8bb29fbdaad044f852fed00ec82/idna-2.10-py2.py3-none-any.whl
imagesize @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/dc/f4/20/8017e7e8c2265d4cd21d6f0288a81425497a5024494edae86dc6d9df79/imagesize-1.2.0-py2.py3-none-any.whl
import-linter @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/ca/90/f4/26fd023850fb7fa51627893105b0693372462b146cf0cb5e34108051e7/import-linter-1.2.1.tar.gz
iniconfig @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/fa/b0/c6/10cfac68c9e6de9d2a1678366ca89fd9292b362c1760dbe758e41691cb/iniconfig-1.1.1-py2.py3-none-any.whl
ipdb @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/91/d9/cc/ce127596ff2468c355bf34d81b146cca3da752b881ea992878990ed49c/ipdb-0.13.4.tar.gz
ipython @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/2a/6e/47/bd496d51c6d11ccff0361c8407f2fa5a6ac0969b7e759e85e99c5a87c6/ipython-7.16.1-py3-none-any.whl
ipython-genutils @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/b4/31/01/6f96480580d1674cab0b5e26dc9fca7bbdf7a2fd5811a7807a92436268/ipython_genutils-0.2.0-py2.py3-none-any.whl
isort @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/13/0e/9d/0ac87b4f86576f57416f5d21432dec16c02955743e7afe51afe253a24b/isort-5.7.0-py3-none-any.whl
jedi @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/2a/5b/d6/62e1f4e7b392c3e7f8258bbe3159dff695814a46e65547cd547ca0fedb/jedi-0.18.0-py2.py3-none-any.whl
Jinja2 @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/c5/c3/19/5cf0bc7ff64d230dd7d23c044479d2b44dabe877ca4165c2d85cbe996d/Jinja2-2.11.3-py2.py3-none-any.whl
jmespath @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/2c/f0/52/b0ba93d941bd49c8719dee7ca27d2096bf96e17948667388c3ee2ac8f8/jmespath-0.10.0-py2.py3-none-any.whl
lark-parser @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/60/68/16/15fa3c28edfd2a2a8029a62c473dca54930759c841551e4988f7be910c/lark_parser-0.11.1-py2.py3-none-any.whl
libcst @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/8b/6e/9c/e339381fc4b7f0f1897d4c093a13df17b7d5cd01f942b6f2208d3a92bc/libcst-0.3.17-py3-none-any.whl
loguru @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/86/46/fb/57abc2a88da92cb957a6f2c96bd3e58662eede7ebc1adce938a5ea46c4/loguru-0.5.3-py3-none-any.whl
lxml @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/34/09/d8/d1a069832f1d066f6227dae5be98d4850015b5c8b4b685be9841742b8e/lxml-4.6.3-cp39-cp39-win_amd64.whl
m2r2 @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/9d/bd/39/c556356a19abe97c5b9fd50723b05832d1db787d14bdf3e75cc9e6dce0/m2r2-0.2.7-py3-none-any.whl
MarkupSafe @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/c2/5a/23/cb884221b19ded365c12c87d599f452879404b6aceb511eec4d5e9be77/MarkupSafe-1.1.1-cp39-cp39-win_amd64.whl
marshmallow @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/0a/a6/1e/da00d5d5d7181dba1448a4998d5f9e49c4081c09e7a097ecbe1fd6679f/marshmallow-3.10.0-py2.py3-none-any.whl
marshmallow-polyfield @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/15/54/1a/bf7b323a7e9bef7044335503b044f792374a95673e74fcf13d80690a10/marshmallow_polyfield-5.10-py3-none-any.whl
mccabe @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/96/5e/5f/21ae5296697ca7f94de4da6e21d4936d74029c352a35202e4c339a4253/mccabe-0.6.1-py2.py3-none-any.whl
mistune @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/33/31/4c/2d69dc65d06d1c8f8b00b8e995e24bae97fce2e1f8ec5d8d2d98e852da/mistune-0.8.4-py2.py3-none-any.whl
more-itertools @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/06/8c/08/25f009f5a9697bfe17bb5789059239edc9053bc9c06ed7db76f146ce57/more_itertools-8.7.0-py3-none-any.whl
mypy @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/3d/2a/a1/be346e4798a5319c7c816d873847992c01bdeea863a113bc5e541fea6a/mypy-0.812-cp39-cp39-win_amd64.whl
mypy-extensions @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/92/45/bf/1807ce854ff668d92602207a37bfa9316def2a3f257bd03c4c5be4bc9b/mypy_extensions-0.4.3-py2.py3-none-any.whl
natsort @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/b7/5b/6a/03c7f99763bed95ea98876477fdb101c0f306c53cc3dc52218b4ba1822/natsort-7.1.1-py3-none-any.whl
networkx @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/47/84/e5/4dc45253033f225419172a1cb84d298fada79643f040b89aca37cf03e1/networkx-2.5-py3-none-any.whl
nitpick @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/67/e2/a9/8e79470c323f14d42df3a2b33b85ecdd7748b55f0c07e3b22e7483e117/nitpick-0.26.0-py3-none-any.whl
packaging @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/c2/76/aa/699ed723689e741411dfbdbc9b8c392079d54fb1e1a1b814df7be7cf47/packaging-20.9-py2.py3-none-any.whl
parso @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/61/0e/11/1101ce07fbc6828dcc0f8951a808bce71c53776aea4788ecf4fe8f0b60/parso-0.8.1-py2.py3-none-any.whl
pbr @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/cc/c3/8b/59e81e4d4dc462a709f431f57734c5e7d971fd97415a857065c3886109/pbr-5.5.1-py2.py3-none-any.whl
pep8-naming @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/a4/93/c7/a3b9b8b4aef682b4caa67015d897aff3d064860a460124ad8a23b6f45f/pep8_naming-0.11.1-py2.py3-none-any.whl
pickleshare @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/b5/48/a1/d2b823337003d531d87cf0d503ef28bb579703a74d14ad24a88863d616/pickleshare-0.7.5-py2.py3-none-any.whl
pluggy @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/29/58/fc/ed8b7451d3ef91a6465024f5656141da996e7aafd4d41a1659629a75e7/pluggy-0.13.1-py2.py3-none-any.whl
prompt-toolkit @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/85/ed/40/229a34dee2081fd935c85f600bfcddb0974a96a88f099412c62b58d4fa/prompt_toolkit-3.0.3-py3-none-any.whl
py @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/60/79/0b/c48bd9c2a989aa8b1eb7a67cd02b053c10734f2e4e5665f7995f09999c/py-1.10.0-py2.py3-none-any.whl
pycodestyle @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/4c/30/97/026c283ef67eb248e5b7e6fad1f8ffb99dae50c11fd93eb939fd7c1f46/pycodestyle-2.7.0-py2.py3-none-any.whl
pydantic @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/f8/b5/38/f5512452381d985aaede15306ecef5e9fc522390ff41e92bb7ab06727d/pydantic-1.7.3-cp39-cp39-win_amd64.whl
pydocstyle @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/f8/8f/c5/799c9286bd88d1d364fb44716368a593e61a8652b827793d9c28bbf6e7/pydocstyle-5.1.1-py3-none-any.whl
pyflakes @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/23/41/2c/087be0ff27d246b7ef3ec86248611398d73cbbea1b33b87f07d69820d3/pyflakes-2.3.0-py2.py3-none-any.whl
Pygments @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/bf/63/90/6f6342dd06f01126b4c490f6da1ddcbcc30899480e797b1e6bb21fc8b7/Pygments-2.8.1-py3-none-any.whl
pyparsing @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/92/0f/cf/effdcd5d76a6186df0969f85b3b030284ff8058936d5016540b5258ea3/pyparsing-2.4.7-py2.py3-none-any.whl
pytest @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/61/a8/ec/1ad0d4114cf9e069c28315cff0a4774573e5b8656efa7a78d2f50c20e4/pytest-6.2.2-py3-none-any.whl
pytest-cov @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/c2/87/fe/0f7becb86948f177058941f4469ac3c20b8c2bddab5e8a11800146c8d9/pytest_cov-2.11.1-py2.py3-none-any.whl
pytest-randomly @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/4d/6d/ce/9788e56a85ba8101173cd5848240c1eff2830eec55d9394a915c6a9b60/pytest_randomly-3.5.0-py3-none-any.whl
python-slugify @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/c7/a7/df/f5e1ea59556a505b43a42b5730c4db933dbde2f994747f2af3dd185fa1/python-slugify-4.0.1.tar.gz
pytz @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/b0/a7/8d/54de3ab4d1ff29abbbca1e9ccbaefdc2a1b290138311b84f73bee16de1/pytz-2021.1-py2.py3-none-any.whl
PyYAML @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/2d/bf/0f/f98d5d175b3387431cd09610047ee5b3cb2cdebb8f1b09a6fd0fc0fafb/PyYAML-5.4.1-cp39-cp39-win_amd64.whl
requests @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/22/0a/9d/0df883fbffbb406d0cddbb35e881e4ac6bfb8f0dee8733056b6a054bf7/requests-2.25.1-py2.py3-none-any.whl
restructuredtext-lint @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/b6/09/80/91d176f17ba9a28291203e41600b294aa26214e185082bcb0cc3543588/restructuredtext_lint-1.3.2.tar.gz
ruamel.yaml @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/e3/77/60/c6e35eb5461cad7856f44e8108e59233a37c83041ce3e4055d79d42476/ruamel.yaml-0.16.12-py2.py3-none-any.whl
safety @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/8f/cc/5a/1dc0196c65cdb3f015bad143c000d159aca6f838a21d7d10872b092b83/safety-1.10.3-py2.py3-none-any.whl
six @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/dd/1c/65/ad0dea11136f5a869f072890a0eea955aa8fc35b90c85c55249fd3abfe/six-1.15.0-py2.py3-none-any.whl
smmap @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/ff/0b/5d/5cb81b7474d0f149717fe4278f98ed66241be438f436b576bd0ca2895f/smmap-3.0.5-py2.py3-none-any.whl
snapshottest @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/41/e6/24/1160e603c1219e4be3367a53b10edba7d146d782e5d5b5dc164f88683f/snapshottest-0.6.0-py2.py3-none-any.whl
snowballstemmer @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/c7/56/66/7613028d4906686fd240574f9e4ec773d99d60753a515f163d21b44935/snowballstemmer-2.1.0-py2.py3-none-any.whl
sortedcontainers @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/b2/00/80/2e76f3e4ccd228a3f48749024777a3970fa07ee5f884d62bf521558e66/sortedcontainers-2.3.0-py2.py3-none-any.whl
Sphinx @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/fc/66/93/d8b346d30433a87426447a82f77f04cc8ece0c195abc7160a41727579f/Sphinx-3.5.3-py3-none-any.whl
sphinx-autodoc-typehints @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/f0/03/c2/c8f159519b299604925e20586389c2367096e7fa09344111f2646f57f6/sphinx_autodoc_typehints-1.11.1-py3-none-any.whl
sphinxcontrib-applehelp @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/8d/eb/86/eec708bb3ff50c9780e78f36a9cb82cd9ff8030a90bd23b9a6f20aecca/sphinxcontrib_applehelp-1.0.2-py2.py3-none-any.whl
sphinxcontrib-devhelp @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/56/a5/74/11ccaa7737f06a10422027e0595b24d243af7a7a1dc4982dec22044c28/sphinxcontrib_devhelp-1.0.2-py2.py3-none-any.whl
sphinxcontrib-htmlhelp @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/5e/a7/c5/bea947b6372fea0245c08ff13ff69dfba86e5b185fc82bac084bb1b7f4/sphinxcontrib_htmlhelp-1.0.3-py2.py3-none-any.whl
sphinxcontrib-jsmath @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/d2/22/96/2076357e64b369910aa24a20d5b719beb24a1487146e4742476ee1e2d8/sphinxcontrib_jsmath-1.0.1-py2.py3-none-any.whl
sphinxcontrib-mermaid @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/f7/89/a8/27378f090fe89dff5cdf5e39cffbe73d8ac5e1d67f056e520a688ecbfd/sphinxcontrib_mermaid-0.6.3-py2.py3-none-any.whl
sphinxcontrib-qthelp @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/32/fc/a9/112a82396d53ec629c1450253a6ded4d94d7ffffd63acd49879543ece9/sphinxcontrib_qthelp-1.0.3-py2.py3-none-any.whl
sphinxcontrib-serializinghtml @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/f9/4e/a9/9d5ca3c1a967e164de9a5fbce38bc5d740c88541c8cf2018595785f2e6/sphinxcontrib_serializinghtml-1.1.4-py2.py3-none-any.whl
stevedore @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/c2/31/a7/c2802b19c1cba9407f4254c97dacd72884d3b27c63bbbb3ada4edbf3a8/stevedore-3.3.0-py3-none-any.whl
termcolor @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/a2/5d/c7/e4ccb3b3bb8d3e3aff995fb6ffb12cfc78bbc8affa283907ee5eb5a5a5/termcolor-1.1.0.tar.gz
testfixtures @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/87/49/bd/0bf2640dd53740f9dc21bb1eecb8093631a71ea4703c8ef1a3a7bdd42d/testfixtures-6.17.1-py2.py3-none-any.whl
text-unidecode @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/34/f9/c2/484c44b08bab89d472229bbd257fcc1d1c6273ee027f01cb08c4e3c309/text_unidecode-1.3-py2.py3-none-any.whl
tokelor @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/01/38/10/c452498a469bf37f62b7065e28334a0a56f0b49ba942ddf20748ec2efc/tokelor-0.1.6-py3-none-any.whl
toml @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/6b/6a/c9/53b19f7870a77d855e8b05ecdc98193944e5d246dafe11bbcad850ecba/toml-0.10.2-py2.py3-none-any.whl
tomlkit @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/6a/39/2a/ebe41a1a0fb862057fa62ec18ed0ed5fa9c8867f707a1a1e3a4c235f46/tomlkit-0.7.0-py2.py3-none-any.whl
traitlets @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/08/5a/74/d32e89818bf605272bd2b53aa4bf254d8c7f562541f7031f1a5d0925fd/traitlets-4.3.3-py2.py3-none-any.whl
typed-ast @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/63/30/46/aad7bd30d645571ce8aa2f192430c51caad7eecbfca1a464675ddd5248/typed_ast-1.4.2-cp39-cp39-win_amd64.whl
typing-extensions @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/ab/c3/72/446cb2c521f10fc837619e8a7c68ed3c3bd74859bd625b7d74f38a159b/typing_extensions-3.7.4.3-py3-none-any.whl
typing-inspect @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/38/a5/13/558baeb3e86a1b962a20af49db4e6ca123ae203fbd234a65fac8ec4b10/typing_inspect-0.6.0-py3-none-any.whl
urllib3 @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/3d/49/75/4245c9a53c80e9d437e00720b38959ccd850e173b62242bcea85c1b100/urllib3-1.26.3-py2.py3-none-any.whl
wcwidth @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/7d/f4/60/0737157bb9711fec72c70dff523aa54491eef317e0d586cf5388ff0908/wcwidth-0.2.5-py2.py3-none-any.whl
wemake-python-styleguide==0.15.2
win32-setctime @ file:///C:/Users/ted/AppData/Local/pypoetry/Cache/artifacts/2c/e5/9c/d5aaac1d96ab96304e0b9e3610d2e1f828ef9d427257e2845af2d92119/win32_setctime-1.0.3-py3-none-any.whl

### OS information

Win10

I would like to work on this. I started and am half way on this.

#1268 is remotely related.

**Comment by maltevesper** [comment]
I dont think the system info I was asked to paste did anyone a favour.
Can this please be assigned to me?

**Comment by sobolevn** [comment]
@maltevesper thanks! 

One note from me: `WPS211` is legit, because, well, functions do have a lot of arguments. It is not dependent on what annotation you use. `WPS428` is also legit. You should use `"""Docstring describing this overload case."""` instead

Related #1779

**Comment by maltevesper** [comment]
Ok, while I can understand your point that WPS428 can and should be worked around with docstrings, I think I would like to clarify WPS211 a bit more:

If I have three overloads I would potentially get four warnings `WPS211`: one for the function itself and up to one for each overload. In a legit use case, I now have to write four `# noqa: WPS211` comments.

This clutters the code with WPS211 comments, and I believe WPS has a violation for too many `# noqa` comments.
I believe a warning for the actual function to be sufficient.

Nevertheless, I shall follow this up with a pull request soon. Then I can present a concrete example. But I won't manage today as I hoped before. For now, a screenshot of the demo (I am missing the test adjustment) has to suffice. I do not see the advantage of being warned twice there.

![image](https://user-images.githubusercontent.com/10866845/112392991-a1407200-8cfa-11eb-9936-f1f0d8746698.png)

**Comment by sobolevn** [comment]
Please, also consider the case with `.pyi` files, where a function does not have the body, only `@overload` cases.

**Comment by sobolevn** [comment]
I removed `WPS211` for overloads.
